app/script/tracking/EventTrackingRepository.js

```diff
@@ -76,44 +76,88 @@ z.tracking.EventTrackingRepository = class EventTrackingRepository {
    */
   init(privacyPreference) {
     this.privacyPreference = privacyPreference;
-    this.logger.info(`Initialize tracking and error reporting: ${this.privacyPreference}`);
-
-    return Promise.resolve()
-      .then(() => {
-        if (this._isDomainAllowedForTracking() && this.privacyPreference) {
-          this._enableErrorReporting();
-          return this._initTracking();
-        }
-        return undefined;
-      })
-      .then(providerInstance => this._initAnalytics(providerInstance))
-      .then(() => amplify.subscribe(z.event.WebApp.PROPERTIES.UPDATE.PRIVACY, this.updatePrivacyPreference));
+    this.logger.info(`Initialize analytics and error reporting: ${this.privacyPreference}`);
+
+    const privacyPromise = this.privacyPreference ? this._enableServices() : Promise.resolve();
+    return privacyPromise.then(() => {
+      amplify.subscribe(z.event.WebApp.PROPERTIES.UPDATE.PRIVACY, this.updatePrivacyPreference);
+    });
   }
 
   updatePrivacyPreference(privacyPreference) {
-    if (privacyPreference !== this.privacyPreference) {
+    const hasPreferenceChanged = privacyPreference !== this.privacyPreference;
+    if (hasPreferenceChanged) {
       this.privacyPreference = privacyPreference;
 
-      if (privacyPreference) {
-        this._enableErrorReporting();
-        if (this._isDomainAllowedForTracking()) {
-          this._reEnableTracking();
-        }
-      } else {
-        this._disableErrorReporting();
-        this._disableTracking();
-      }
+      return this.privacyPreference
+        ? this._enableServices().then(() => this._trackEvent(z.tracking.EventName.SETTINGS.OPTED_IN_TRACKING))
+        : this._trackEvent(z.tracking.EventName.SETTINGS.OPTED_OUT_TRACKING).then(() => this._disableServices());
```

Comment:
In order not to create a single point of failure with whatever tracking system we will use in the future, let's not make the call to `_disableServices` dependent on the result of tracking the event